common/content/statusline.js

```diff
@@ -127,17 +127,13 @@ const StatusLine = Module("statusline", {
 
     // set the visibility of the statusline
     setVisibility: function (request) {
-        if ( typeof this.setVisibility.currVisibility == 'undefined' ) {
-            this.setVisibility.currVisibility = true;
-            this.setVisibility.prevVisibility = true;
-            this.setVisibility.contentSeparator
-                = highlight.get('ContentSeparator').value;
-
-            // kinds of requests
-            this.setVisibility.MODE_ON  = 0;      // commandline active
```

Comment:
I don't think that removing `TOGGLE` is a big problem. This is a really new feature that isn't used by many users probably. Vimperator is under heavy development and design changes and I'd always expect incompatibilities.
Nevertheless I think a function to cycle between all tree modes would be good and I'll see if there's a good and simple way to implement it (maybe by adding `TOGGLE` again).

`UPDATE` and `TOGGLE` have different behavior.

- `TOGGLE` changes the mode on each call so that you can cycle through all modes (before my changes you switched out of `AUTO` into `ON` or `OFF` and there was no way to get back to `AUTO`).
- `UPDATE` on the other hand can be called multiple times without any further effects. It will apply the current configuration (i.e. value of `statuslinevisibility` and `window.fullscreen`). This is necessary in situations when one of these properties changes (`:set statuslinevisibility=`, entering or leaving fullscreen).